src/command_modules/azure-cli-resource/tests/test_locks.py

```diff
@@ -0,0 +1,39 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import unittest
+
+from azure.cli.testsdk import ScenarioTest, ResourceGroupPreparer
+
+
+class StorageAccountTests(ScenarioTest):
+    def test_list_locks(self):
+        locks_list = self.cmd('az lock list').get_output_in_json()
+        assert len(locks_list) == 0
+
+    def test_lock_create_list_delete(self):
+        for lock_type in ['ReadOnly', 'CanNotDelete']:
+            self.cmd('az lock create -n foo --lock-type {}'.format(lock_type))
+            locks_list = self.cmd('az lock list').get_output_in_json()
+            assert len(locks_list) > 0
+            lock = self.cmd('az lock show -n foo').get_output_in_json()
+            self.assertEqual(lock.get('name', None), 'foo')
+            self.assertEqual(lock.get('level', None), lock_type)
+            self.cmd('az lock delete -n foo')
+
+    @ResourceGroupPreparer()
+    def test_lock_create_list_delete_resource_group(self, resource_group):
+        for lock_type in ['ReadOnly', 'CanNotDelete']:
+            self.cmd('az lock create -n foo -g {} --lock-type {}'.format(resource_group, lock_type))
+            locks_list = self.cmd('az lock list').get_output_in_json()
```

Comment:
It seems that the test for the lock scenario should be something like this:
1. Create the lock on the foo.
2. Attempt to update/delete the foo and verify that fails because the lock is in place.
3. Delete the lock from the foo.
4. Update/delete the foo successfully since the lock was removed.

This would be preferable because in the past we've had the situation where someone was trying to create a resource lock and it "succeeded" by actually creating a resource group lock.  This type of test should ensure that this type of logic error doesn't occur due to a regression.